### PR TITLE
solution was created

### PR DESCRIPTION
**Pull request by vlad-ptashnichenko**
[DEMO LINK](https://vlad-ptashnichenko.github.io/react_tabs-js/)

### CODE REVIEW

File: src/components/Tabs/Tabs.jsx [hunk 1]
```diff
@@ -1 +1,49 @@
...
+              )}
+            </li>
+          ))}
+        </ul>
+      </div>
+
+      <div className="block" data-cy="TabContent">
+        {tabs.reduce((text, tab) => (
+          tab.id === selectedTabId ? tab.content : text
+        ), '')}
```
Comment (etojeDenys) [inline src/components/Tabs/Tabs.jsx:45]: we can just use selectTab instead of reduce

**Review by etojeDenys — CHANGES_REQUESTED** [review]

File: src/components/Tabs/Tabs.jsx [hunk 2]
```diff
@@ -1 +1,47 @@
...
+              ) : (
+                <a
+                  href={`#${tab.id}`}
+                  data-cy="TabLink"
+                  onClick={() => onTabSelected(tab)}
+
+                >
+                  {tab.title}
+                </a>
+              )}
```
Comment (etojeDenys) [inline src/components/Tabs/Tabs.jsx:36]: same here, we can just move this condition into the onClick event

**Review by etojeDenys — CHANGES_REQUESTED** [review]

File: src/App.jsx [hunk 3]
```diff
@@ -1,40 +1,34 @@
...
-          <li className="is-active" data-cy="Tab">
-            <a href="#tab-1" data-cy="TabLink">Tab 1</a>
-          </li>
-
-          <li data-cy="Tab">
-            <a href="#tab-2" data-cy="TabLink">Tab 2</a>
-          </li>
+export const App = () => {
+  const [selectedTabId, setSelectedTabId] = useState(tabs[0].id);
+  const [selectedTabTitle, setSelectedTabTitle] = useState(tabs[0].title);
```
Comment (etojeDenys) [inline src/App.jsx:16]: looks like redundant state, we can try to find the title using the `tabs` and `selectedTabId`

**Review by etojeDenys — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 3 above)
Comment (Viktor-Kost) [inline src/App.jsx:16]:
```suggestion
  const [selectedTab, setSelectedTab] = useState(tabs[0]);
```

File: src/App.jsx [hunk 4]
```diff
@@ -1,40 +1,34 @@
...
 
-          <li data-cy="Tab">
-            <a href="#tab-3" data-cy="TabLink">Tab 3</a>
-          </li>
-        </ul>
-      </div>
+  return (
+    <div className="section">
+      <h1 className="title">
+        {`Selected tab is ${selectedTabTitle}`}
```
Comment (Viktor-Kost) [inline src/App.jsx:21]:
```suggestion
        {`Selected tab is ${selectedTab.title}`}
```

File: src/App.jsx [hunk 5]
```diff
@@ -1,40 +1,34 @@
...
+      </h1>
 
-      <div className="block" data-cy="TabContent">
-        Some text 1
-      </div>
+      <Tabs
+        tabs={tabs}
+        selectedTabId={selectedTabId}
```
Comment (Viktor-Kost) [inline src/App.jsx:26]:
```suggestion
        selectedTabId={selectedTab.id}
```

File: src/App.jsx [hunk 6]
```diff
@@ -1,40 +1,34 @@
...
+        onTabSelected={(tab) => {
+          setSelectedTabTitle(tab.title);
+          setSelectedTabId(tab.id);
+        }}
```
Comment (Viktor-Kost) [inline src/App.jsx:30]:
```suggestion
        onTabSelected={setSelectedTab}
```

File: src/components/Tabs/Tabs.jsx [hunk 7]
```diff
@@ -1 +1,39 @@
...
+              data-cy="Tab"
+              key={tab.id}
+              className={cn({ 'is-active': selectTab.id === tab.id })}
+            >
+              <a
+                href={`#${tab.id}`}
+                data-cy="TabLink"
+                onClick={() => (
+                  selectTab.id === tab.id ? '' : onTabSelected(tab)
+                )}
```
Comment (Viktor-Kost) [inline src/components/Tabs/Tabs.jsx:25]: move to handler
```
const handleTabSelect = (tab) => () => {
   if(tab.id === selectedTabId) {
     return
   }
   
   onTabSelected(tab)
}
```
```suggestion
                onClick={handleTabSelect(tab)}
```

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

**Pull request closed without merging**